source/common/upstream/logical_dns_cluster.cc

```diff
@@ -21,8 +21,7 @@ LogicalDnsCluster::LogicalDnsCluster(const Json::Object& config, Runtime::Loader
   }
 
   dns_url_ = hosts_json[0]->getString("url");
-  Network::Utility::hostFromTcpUrl(dns_url_);
-  Network::Utility::portFromTcpUrl(dns_url_);
+  hostname_ = Network::Utility::hostAndPortFromTcpUrl(dns_url_);
```

Comment:
According to the RFC, 
```A "host" without any trailing port information implies the default port for the service requested (e.g., "80" for an HTTP URL). ```

I assumed that if its not a default port, then the port must be part of the Host header.
If stuff works without a port, I would be more than happy to reuse the hostFromTcpUrl function. 

@kyessenov thoughts? You hit some issues in k8s regarding this right?